### PR TITLE
[Type Hints] VAE models

### PR DESCRIPTION
**Pull request by daspartho**
Based on 3, 4, 7, 10, and 11 of issue https://github.com/huggingface/diffusers/issues/287

@anton-l could you check it?
Thanks :)

### CODE REVIEW

File: src/diffusers/models/vae.py [hunk 1]
```diff
@@ -293,7 +295,7 @@ def __init__(self, parameters, deterministic=False):
         if self.deterministic:
             self.var = self.std = torch.zeros_like(self.mean).to(device=self.parameters.device)
 
-    def sample(self, generator=None):
+    def sample(self, generator: Optional[torch.Generator] = None):
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:298]:
```suggestion
    def sample(self, generator: Optional[torch.Generator] = None) -> torch.FloatTensor:
```

File: src/diffusers/models/vae.py [hunk 2]
```diff
@@ -382,7 +384,7 @@ def decode(self, h, force_not_quantize=False):
         dec = self.decoder(quant)
         return dec
 
-    def forward(self, sample):
+    def forward(self, sample: torch.FloatTensor):
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:387]:
```suggestion
    def forward(self, sample: torch.FloatTensor) -> torch.FloatTensor:
```

File: src/diffusers/models/vae.py [hunk 3]
```diff
@@ -440,7 +442,7 @@ def decode(self, z):
         dec = self.decoder(z)
         return dec
 
-    def forward(self, sample, sample_posterior=False):
+    def forward(self, sample: torch.FloatTensor, sample_posterior: bool = False):
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:445]:
```suggestion
    def forward(self, sample: torch.FloatTensor, sample_posterior: bool = False) -> torch.FloatTensor:
```

**Review by patrickvonplaten — COMMENTED** [review]
Thanks a lot @daspartho! Left a couple of suggestions to also return the return type :-)

**Review by patrickvonplaten — APPROVED** [review]

**Comment by daspartho** [comment]
@patrickvonplaten applied the suggestions :)

**Review by patrickvonplaten — APPROVED** [review]
Great - looks nice!

**Pull request merged**